service/src/main/java/bio/terra/workspace/common/utils/ControllerValidationUtils.java

```diff
@@ -84,4 +92,13 @@ public static void validateCloudPlatform(ApiCloudPlatform platform) {
             "Invalid cloud platform. Currently, only AZURE and GCP are supported.");
     }
   }
+
+  public static void validateUserFacingId(String userFacingId) {
```

Comment:
do we have tests for these, or that the validation runs?

Reply:
I tested with Swagger. Unit tests wouldn't exercise this code (since unit test calls service directly). Didn't seem worth making an integration test for this. (And there isn't a ControllerValidationUtils.java for this file. I could make one, but the other parts of this file aren't being unit tested.)

Reply:
I'd suggest adding tests in `WorkspaceLifecycle.java` integration test. Right now it has no negative cases in it at all which seems like a problem. In general, our integration tests are not doing a good job at testing invalid inputs.

Reply:
Done.

Just curious, is it possible to write a unit test that calls controller (workspaceApi?) instead of service?

Reply:
I don't know of any way, but that does not mean there isn't one.